src/cargo.rs

```diff
@@ -250,14 +250,21 @@ impl Root {
     }
 }
 
-pub fn root(mode: XargoMode) -> Result<Option<Root>> {
+pub fn root(mode: XargoMode, path: Option<&str>) -> Result<Option<Root>> {
     // Don't require a 'Cargo.toml' to exist when 'xargo-check' is used
     let name = match mode {
         XargoMode::Build => "Cargo.toml",
         XargoMode::Check => "Xargo.toml"
     };
-    let cd = env::current_dir().chain_err(|| "couldn't get the current directory")?;
 
+    let cd = match path {
+        None => env::current_dir().chain_err(|| "couldn't get the current directory")?,
+        Some(p) => {
+            let mut pb = PathBuf::from(p);
+            pb.pop();
```

Comment:
Why do you need to pop once?

Reply:
pb is the path to `Cargo.toml`, popping gives the directory.

Reply:
`pb` comes from `args.root()` which you documented to be the root *directory*... something seems off here?